### PR TITLE
feat: enable group condition

### PR DESCRIPTION
**Pull request by zoli**
As part of #2734 this enables group configuration. So grouping by a field can also have more options such as group condition. For example when grouping by a date field type the user can choose to group by day, month, week, etc.

### Feature Preview

https://github.com/AppFlowy-IO/AppFlowy/assets/3286303/32e75359-6b47-4c28-888c-265894d27935

#### PR Checklist

- [x] My code adheres to [AppFlowy's Conventions](https://docs.appflowy.io/docs/documentation/software-contributions/conventions)
- [x] I've listed at least one issue that this PR fixes in the description above.
- [ ] I've added a test(s) to validate changes in this PR, or this PR only contains semantic changes.
- [x] All existing tests are passing.

### CODE REVIEW

**Review by skydiver1980 — APPROVED** [review]

**Comment by richardshiue** [comment]
My biggest concern with this PR is the condition inside GroupSettings. Using an int to store the condition (which is why I guess the DateCondition is removed?) leads to confusion between different field types. I think it would be better to specify the enum type in each`GroupCustomize` implementation.

File: frontend/appflowy_flutter/lib/plugins/database_view/application/setting/group_bloc.dart [hunk 1]
```diff
@@ -92,8 +106,10 @@ class DatabaseGroupEvent with _$DatabaseGroupEvent {
   const factory DatabaseGroupEvent.initial() = _Initial;
   const factory DatabaseGroupEvent.setGroupByField(
     String fieldId,
-    FieldType fieldType,
-  ) = _DatabaseGroupEvent;
+    FieldType fieldType, {
+    int? condition,
+    bool? hideEmpty,
+  }) = _DatabaseGroupEvent;
```
Comment (richardshiue) [inline frontend/appflowy_flutter/lib/plugins/database_view/application/setting/group_bloc.dart:112]: You can use optional positional parameters here with `[...]`to maintain consistency

Comment (zoli) [reply frontend/appflowy_flutter/lib/plugins/database_view/application/setting/group_bloc.dart]: Done.

**Comment by zoli** [comment]
> But I think the int should only be seen when converting to and from PBs, and when reading/saving from Collab. The `GroupSettings` shouldn't deal with a single int and the frontend looks like it's dealing with the base ProtobufEnum.

I don't get your point exactly. Do you mean to deal with it like `filters` which stores the data as `Vec<u8>`? If yes I don't know how it's different from storing an int.
Personally, I like avoiding match patterns based on field type and creating struct based on each case when its possible.

**Pull request closed without merging**